--- /src/plugin.py	2019-10-20 11:52:06.000000000 +0200
+++ /src/plugin.py	2019-10-20 20:46:36.968276676 +0200
@@ -2,7 +2,7 @@
 #created by Vasiliks 11.2015
 #added picon providers 05.01.2018
 #added picon IPTVproviders 10.10.2019
-#r0.1_r7  PLI version
+#r0.1_r7  BlackHole version
 from . import _
 from Components.ActionMap import ActionMap
 from Components.config import config, ConfigEnableDisable, ConfigInteger, ConfigSelection, ConfigSubsection, getConfigListEntry
@@ -10,7 +10,7 @@
 from Components.Label import Label
 from Components.MenuList import MenuList
 from Components.MultiContent import MultiContentEntryProgress, MultiContentEntryText, MultiContentEntryPixmapAlphaBlend
-from enigma import BT_SCALE, getDesktop, eListboxPythonMultiContent, eServiceCenter, eServiceReference, eTimer, gFont, loadPNG, RT_HALIGN_LEFT, RT_HALIGN_CENTER, RT_HALIGN_RIGHT, RT_WRAP, RT_VALIGN_CENTER
+from enigma import getDesktop, eListboxPythonMultiContent, eServiceCenter, eServiceReference, eTimer, gFont, ePicLoad, RT_HALIGN_LEFT, RT_HALIGN_CENTER, RT_HALIGN_RIGHT, RT_WRAP, RT_VALIGN_CENTER
 from Components.ParentalControl import parentalControl
 from Plugins.Plugin import PluginDescriptor
 from Screens.ChannelSelection import ChannelSelection
@@ -56,7 +56,6 @@
 config.plugins.vZapHistory.duratcolor = ConfigSelection(default="0x8f8f8f", choices = color)
 config.plugins.vZapHistory.duratcolor_sel = ConfigSelection(default="0xffff55", choices = color)
 config.plugins.vZapHistory.barcolor = ConfigSelection(default="0x0099ff", choices = color)
-config.plugins.vZapHistory.barcolor_sel = ConfigSelection(default="0xffff55", choices = color)
 
 skin_sd = """
     <screen name="vZapHistory" position="center,center" size="530,400" >
@@ -174,7 +173,6 @@
     duratcolor = int(config.plugins.vZapHistory.duratcolor.value, 16)
     duratcolor_sel = int(config.plugins.vZapHistory.duratcolor_sel.value, 16)
     barcolor = int(config.plugins.vZapHistory.barcolor.value, 16)
-    barcolor_sel = int(config.plugins.vZapHistory.barcolor_sel.value, 16)
 
     if screenWidth == 1920:
         if config.plugins.vZapHistory.viewMode.value == 'pic_serv':
@@ -287,14 +285,14 @@
         bar_size=(140, 8)
 
     if 'pic_serv' in config.plugins.vZapHistory.viewMode.value:                                                
-        res.append(MultiContentEntryPixmapAlphaBlend(pos=png_pos, size=png_size, flags = BT_SCALE, png=png))
+        res.append(MultiContentEntryPixmapAlphaBlend(pos=png_pos, size=png_size,  png=png))
     if 'pic_serv_prov' in config.plugins.vZapHistory.viewMode.value:                                          
-        res.append(MultiContentEntryPixmapAlphaBlend(pos=png_prov_pos, size=png_prov_size, flags = BT_SCALE, png=png_prov))
+        res.append(MultiContentEntryPixmapAlphaBlend(pos=png_prov_pos, size=png_prov_size, png=png_prov))
     res.append(MultiContentEntryText(pos=serviceName_pos, size=serviceName_size, font=serviceName_font, flags=lasflags, text=serviceName, color=namecolor, color_sel=namecolor_sel))
     res.append(MultiContentEntryText(pos=eventName_pos, size=eventName_size, font=eventName_font, flags=lasflags | RT_VALIGN_CENTER | RT_WRAP, text=eventName, color=eventcolor, color_sel=eventcolor_sel))
     res.append(MultiContentEntryText(pos=durationTime_pos, size=durationTime_size, font=durationTime_font, flags=lasflags, text=durationTime, color=duratcolor, color_sel=duratcolor_sel))
     if bar != 0:
-        res.append(MultiContentEntryProgress(pos=bar_pos, size=bar_size, percent=bar, borderWidth=1, foreColor=barcolor, foreColorSelected=barcolor_sel))
+        res.append(MultiContentEntryProgress(pos=bar_pos, size=bar_size, percent=bar, borderWidth=1, foreColor=barcolor))
     return res
 
 class vZapHistory(Screen, ProtectedScreen):
@@ -414,8 +412,18 @@
                 ref = x[1]
             else:
                 ref = x[2]
-            png = loadPNG(self.findPiconServ(ref))
-            png_prov = loadPNG(self.findPiconProv(ref))
+            self.piconServ = ePicLoad()
+            self.piconProv = ePicLoad()
+            if screenWidth == 1920:
+                self.piconServ.setPara((100, 60, 1, 1, False, 1, '#000f0f0f'))
+                self.piconProv.setPara((100, 60, 1, 1, False, 1, '#000f0f0f'))
+            else:
+                self.piconServ.setPara((90, 54, 1, 1, False, 1, '#000f0f0f'))
+                self.piconProv.setPara((90, 54, 1, 1, False, 1, '#000f0f0f'))
+            self.piconServ.startDecode(self.findPiconServ(ref), 0, 0, False)
+            self.piconProv.startDecode(self.findPiconProv(ref), 0, 0, False)
+            png = self.piconServ.getData()
+            png_prov = self.piconProv.getData()
             info = self.serviceHandler.info(ref)
             if info:
                 name = info.getName(ref).replace('\xc2\x86', '').replace('\xc2\x87', '')
@@ -590,8 +598,7 @@
          getConfigListEntry(_('Progress bar color:'), config.plugins.vZapHistory.barcolor),
          getConfigListEntry(_('Selected service name color:'), config.plugins.vZapHistory.namecolor_sel),
          getConfigListEntry(_('Selected event color:'), config.plugins.vZapHistory.eventcolor_sel),
-         getConfigListEntry(_('Selected duration time color:'), config.plugins.vZapHistory.duratcolor_sel),
-         getConfigListEntry(_('Selected progress bar color:'), config.plugins.vZapHistory.barcolor_sel)])
+         getConfigListEntry(_('Selected duration time color:'), config.plugins.vZapHistory.duratcolor_sel)])
         self['actions'] = ActionMap(['OkCancelActions', 'ColorActions'],
         {'ok': self.save, 'green': self.save,
          'cancel': self.exit, 'red': self.exit}, -2)
